contracts/handlers/ERC20Handler.sol

```diff
@@ -113,15 +113,15 @@ contract ERC20Handler is IDepositHandler, ERC20Safe {
             amount                      := mload(add(data, 0x40))
 
             tokenID                     := mload(0x40)
-            let lenTokenID              := mload(add(0x40, data))
+            let lenTokenID              := mload(add(0x60, data))
```

Comment:
This is because the first 20 are for the chainID? cant we get a comment :)

Reply:
no it's because amount spans from `0x40 - 0x60`. So we've place teh length declaration of the `tokenID` as a uint256 from `0x60 - 0x80` then the actual `bytes tokenId` is located from 0x80 to the end of calldata. We need this to shift the free memory pointer to behind tokenID avoid getting tokenID overwritten after it's been copied to.

Reply:
ignore